modules/client/src/client-common/context.ts

```diff
@@ -211,48 +232,61 @@ export class Context {
    *
    * @var ipfs
    *
-   * @returns {IpfsClient[] | undefined}
+   * @returns {IpfsClient[]}
    *
    * @public
    */
-  get ipfs(): IpfsClient[] | undefined {
-    return this.state.ipfs || defaultState.ipfs;
+  get ipfs(): IpfsClient[] {
+    return this.state.ipfs;
   }
 
   /**
    * Getter for the GraphQL client
    *
    * @var graphql
    *
-   * @returns {GraphQLClient[] | undefined}
+   * @returns {GraphQLClient[]}
    *
    * @public
    */
-  get graphql(): GraphQLClient[] | undefined {
-    return this.state.graphql || defaultState.graphql;
+  get graphql(): GraphQLClient[] {
+    return this.state.graphql;
   }
 
-  // DEFAULT CONTEXT STATE
-  static setDefault(params: Partial<ContextParams>) {
-    if (params.daoFactoryAddress) {
-      defaultState.daoFactoryAddress = params.daoFactoryAddress;
+  // INTERNAL HELPERS
+  private static resolveNetwork(
+    networkish: Networkish,
+    ensRegistryAddress?: string,
+  ): Network {
+    const network = getNetwork(networkish);
+    const networkName = network.name as SupportedNetworks;
+    if (!SupportedNetworksArray.includes(networkName)) {
+      throw new UnsupportedNetworkError(networkName);
     }
-    if (params.signer) {
-      defaultState.signer = params.signer;
+    if (ensRegistryAddress) {
+      if (!isAddress(ensRegistryAddress)) {
+        console.warn("invalid address, using default");
+        // throw new InvalidAddressError()
+      } else {
+        network.ensAddress = ensRegistryAddress;
+      }
     }
+    if (!network.ensAddress) {
+      const ensAddress = LIVE_CONTRACTS[networkName].ensRegistry;
+      if (!ensAddress) {
+        throw new EnsUnsuportedByNetworkError(networkName);
+      }
+      network.ensAddress = ensAddress;
+    }
```

Comment:
Why not combining these 2 blocks into a single one?

Reply:
We cannot combine it it would change the code works:

1. The network object is generated and can contain an `ensAddress`
2. Check if the ensRegistryAddress is specified
3. If its specified and is incorrect, throw an error
4. If is correct, override the value in `network.ensAddress`

If we try to unify this we will override the value of `network.ensAddress` even when the `ensRegistryAddress` is not specified